chef.py

```diff
@@ -149,20 +156,84 @@ def get_absolute_path(path):
     return urllib.parse.urljoin('http://www.' + DOMAIN, path)
 
 
+def make_request(url):
+    response = session.get(url)
+    if response.status_code != 200:
+        LOGGER.error("NOT FOUND: %s" % (url))
+    return response
+
+
 def get_page(path):
     url = get_absolute_path(path)
-    resp = session.get(url)
+    resp = make_request(url)
     return BeautifulSoup(resp.content, 'html.parser')
 
 
 def get_source_id(path):
-    return path.strip("/").split("/")[-1]
+    return path.strip('/').split('/')[-1]
 
 
 def get_content_link(content):
+    """The link to a content has an onclick attribute that executes
+    the res_click function. This function has 4 parameters:
+    - The main file (e.g. an mp4 file, an entry html page to a game).
+    - The type of resource (video, internal link, ...).
+    - A description.
+    - A master file (e.g. for a game, it is a zip file).
+    """
     link = content.find('a', {'id': 'navigate'})
-    regex = re.compile(r"res_click\('(.*)','.*','.*','.*'\)")
+    regex = re.compile(r"res_click\('(.*)','.*','.*','(.*)'\)")
     match = regex.search(link['onclick'])
     link = match.group(1)
-    link = get_absolute_path(link)
-    return link
+    main_file = get_absolute_path(link)
+    master_file = match.group(2)
+    if master_file:
+        master_file = get_absolute_path(master_file)
+    return main_file, master_file
+
+
+def get_zip_file(zip_file_url, main_file, title):
+    """HTML games are provided as zip files, the entry point of the game is
+     main_file. main_file needs to be renamed to index.html to make it
+     compatible with Kolibri.
+    """
+    destpath = tempfile.mkdtemp()
+    try:
+        download_file(zip_file_url, destpath, request_fn=make_request)
+
+        zip_filename = zip_file_url.split('/')[-1]
+        zip_basename = zip_filename.rsplit('.', 1)[0]
+        zip_folder = os.path.join(destpath, zip_basename)
```

Comment:
Another option would be to pass a static `filename='original.zip'` argument to `download_file`, above, so you know what the filename will be. But this works too!